nautobot/ipam/models.py

```diff
@@ -660,10 +680,17 @@ def save(self, *args, **kwargs):
             if self._location is not None:
                 self.location = self._location
 
-        # Determine the subnets and reparent them to this prefix.
-        self.reparent_subnets()
-        # Determine the child IPs and reparent them to this prefix.
-        self.reparent_ips()
+        # Only reparent subnets and ips if any of these fields has been updated.
```

Comment:
Can we add some unit testing around this? I know it's feature creep (sorry!) but I think it's important to test.